gradle-jdks/src/main/java/com/palantir/gradle/jdks/flow/ToolchainFlowAction.java

```diff
@@ -0,0 +1,97 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.gradle.api.flow.BuildWorkResult;
+import org.gradle.api.flow.FlowAction;
+import org.gradle.api.flow.FlowParameters;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ToolchainFlowAction implements FlowAction<Parameters> {
+
+    private static final Logger log = LoggerFactory.getLogger(ToolchainFlowAction.class);
+
+    interface Parameters extends FlowParameters {
+        @Input
+        Property<BuildWorkResult> getBuildResult();
+
+        @Input
+        ListProperty<String> getConfiguredJavaMajorVersions();
+    }
+
+    @Override
+    @SuppressWarnings("LineLength")
+    public void execute(Parameters parameters) {
+        parameters.getBuildResult().get().getFailure().ifPresent(failure -> {
+            List<Throwable> noToolchainsAvailable = Throwables.getCausalChain(failure).stream()
+                    .filter(throwable ->
+                            throwable instanceof org.gradle.jvm.toolchain.internal.NoToolchainAvailableException)
+                    .collect(Collectors.toList());
+            if (noToolchainsAvailable.isEmpty()) {
+                return;
+            }
+            List<String> missingToolchains = noToolchainsAvailable.stream()
+                    .map(exception -> {
+                        Pattern pattern = Pattern.compile("languageVersion=(\\d+)");
+                        Matcher matcher = pattern.matcher(exception.getMessage());
+                        if (matcher.find()) {
+                            return Optional.of(matcher.group(1));
+                        }
+                        return Optional.<String>empty();
+                    })
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toList());
+            String maybeMissingToolchains = missingToolchains.isEmpty()
+                    ? "some java versions"
+                    : String.format("the java versions=%s", missingToolchains);
+            String includeVersionsOption = missingToolchains.isEmpty()
+                    ? "-includeVersion=<newJavaMajorVersion>"
+                    : missingToolchains.stream()
+                            .map(version -> String.format("--includeVersion=%s", version))
+                            .collect(Collectors.joining(" "));
+            log.error(
+                    "\n"
+                        + "\u001B[31m****************************************************************************************************\n"
+                        + "****************************************************************************************************\n"
```

Comment:
Nit: can we make it such that the text fits in the asterisks pls 🙈

Reply:
addressed this in https://github.com/palantir/gradle-jdks/pull/479/commits/4b8a2513668b586ad922dd6869f3e79c238f228b#diff-c7b32bc0b052f8a5cc11b4c0eed62a0c270c15a432e2a52dffe6bc0678df45f6R80: 
![Screenshot 2025-04-08 at 12 09 33 PM](https://github.com/user-attachments/assets/8ea69a5e-4609-47fa-a4c6-6a2bf821421f)